tracer/test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -425,6 +427,11 @@ private protected MockTracerResponse HandleHttpRequest(MockHttpParser.MockHttpRe
                 HandlePotentialDataStreams(request);
                 response = "{}";
             }
+            else if (request.PathAndQuery.StartsWith("/evp_proxy/v1/"))
```

Comment:
I saw somewhere (I can't find it now so I'm commenting here instead) that you remove this endpoint to test with an agent that doesn't support it. Not now, but in the future we should think of how we can make `MockTracerAgent` more configurable. Maybe the ability to enable/disable specific endpoints, or a setting to emulate a specific agent version (which would enable/disable specific endpoints or emulate specific behavior even within the same endpoint).

Reply:
Yes I totally agree with that. In order to test the APM writer callback I had to remove the endpoint from the mock manually.

Reply:
We can already control which endpoints the agent _advertises_ as available by changing `AgentConfiguration.Endpoints`. We _could_ use that to explicitly remove some of these code paths too. That would help us catch cases where sending requests to an agent when it explicitly doesn't support them, though in practice, it should never happen!